packages/transactions/src/OutboundShipment/ListView/ListView.tsx

```diff
@@ -28,21 +27,61 @@ import {
   ChevronDown,
   Tools,
   getNameAndColorColumn,
+  useWindowDimensions,
+  useTheme,
+  useAppBarRectStore,
 } from '@openmsupply-client/common';
 
 import { listQueryFn, deleteFn, updateFn } from '../../api';
 
+const useListViewQueryParams = (initialSortBy: SortingRule<Transaction>[]) => {
+  const { height } = useAppBarRectStore();
+  const { height: windowHeight } = useWindowDimensions();
+  const theme = useTheme();
+  const { mixins } = theme;
+
+  const dataRowHeight = mixins.table.dataRow.height;
+  const headerRowHeight = mixins.table.headerRow.height;
+  const paginationRowHeight = mixins.table.paginationRow.height;
+
+  const numberOfRows = Math.floor(
+    (windowHeight - (height ?? 0) - headerRowHeight - paginationRowHeight) /
+      dataRowHeight
+  );
+
+  const [first, setFirst] = useState(numberOfRows);
+  const [offset, setOffset] = useState(0);
+  const [sortBy, setSortBy] = useState(initialSortBy);
+  const [page, setPage] = useState(0);
+
+  useEffect(() => {
+    setOffset(page * first);
+  }, [first, page]);
+
+  return {
+    first,
+    offset,
+    sortBy,
+    page,
+    setPage,
+    setFirst,
+    setOffset,
+    setSortBy,
+    numberOfRowsToRender: numberOfRows,
+  };
+};
+
 export const OutboundShipmentListView: FC = () => {
+  const { first, offset, sortBy, setSortBy, setPage, numberOfRowsToRender } =
+    useListViewQueryParams([{ id: 'name', desc: false }]);
+
   const { appBarButtonsRef } = useHostContext();
   const { info, success, warning } = useNotification();
 
-  const [queryProps, setQueryProps] = useState<QueryProps<Transaction>>({
-    first: 10,
-    offset: 0,
-  });
   const { data: response, isLoading } = useQuery(
-    ['transaction', 'list', queryProps],
-    () => listQueryFn(queryProps)
+    ['transaction', 'list', { first, offset, sortBy }],
+    () => listQueryFn({ first, offset, sortBy }),
+    { keepPreviousData: true }
```

Comment:
this is stopping `loading` become true and instead just replaces the `data` with the new data which can make it feel a lot nicer, but could feel a lot worse if not on a LAN!